ca/ca.go

```diff
@@ -165,6 +169,9 @@ func (ca *CA) Init(cfg *config.Config) (*CA, error) {
 		return nil, err
 	}
 
+	// TODO does this config have a renewing cert with client auth?
+	webhookTransport.TLSClientConfig = tlsConfig
```

Comment:
No, this config does not have a client certificate, only a server one.

Reply:
We should have a custom tlsConfig for the webhook client, using `GetClientCertificate` and `RootCAs` with the system roots and the server ones. The system roots should not be part of the server ClientCAs.

Reply:
a30444b

Reply:
This tlsConfig is intended for servers, it doesn't include a valid RootCAs. We should allow RootCAs for Web PKI (`x509.SystemCertPool()`) and our CA root.

Reply:
I prefer a different `tls.Config` for clients. They should share the certificate renewer, though.

Reply:
627ee201